rtt/src/channel.rs

```diff
@@ -58,13 +58,20 @@ impl Channel {
         ptr: u32,
         mem: &[u8],
     ) -> Result<Option<Channel>, Error> {
-        let buffer_ptr: u32 = mem.pread_with(Self::O_BUFFER_PTR, LE).unwrap();
+        let buffer_ptr: u32 = match mem.pread_with(Self::O_BUFFER_PTR, LE) {
+            Ok(buffer_ptr) => buffer_ptr,
+            Err(error) => return Err(Error::Scroll(error)),
```

Comment:
I think this should be a an `unwrap` or even better `expect` and be checked on the caller of `from`, such that we can get rid of this error variant (`Scroll`) as it is not meaningful to the user of `probe-rs-rtt` :)

Reply:
Will do this. Thank you for the additional explanation / rationalization in our Matrix chat :)